packages/nimble-components/src/wafer-map/modules/wafer-map-validator.ts

```diff
@@ -0,0 +1,43 @@
+import type { WaferMap } from '..';
+import type { WaferMapValidity } from '../types';
+
+/**
+ * Helper class for the nimble-table to validate that the table's configuration
+ * is valid and report which aspects of the configuration are valid or invalid.
+ */
+export class WaferMapValidator {
+    private invalidGridDimensions = false;
+
+    public constructor(private readonly wafermap: WaferMap) {}
+    public getValidity(): WaferMapValidity {
+        return {
+            invalidGridDimensions: this.invalidGridDimensions
+        };
+    }
+
+    public isValid(): boolean {
+        return Object.values(this.getValidity()).every(x => x === false);
+    }
+
+    public validateGridDimensions(): boolean {
```

Comment:
Could you please elaborate a little bit on the validation logic ? Unless I am missing something it does feel incomplete - there are a couple of cases which might need to be addressed, given that this is public API.   What if `gridMinX`  or  `gridMinY` is negative ?  `gridMinX` is negative and `gridMaxX` is positive ? Can the sign change based on the wafer orientation ? 
What if  `gridMaxX - gridMinX` is less than the die dimension ?

Reply:
Could you also create a test file for the validator that covers all these cases?

Reply:
We support negative inputs, just as we support negative die coordinates.
These grid inputs refer to the virtual grid of dies that are provided as input in an array of points and values. they can have negative coordinates or range from negative to positive. We are mapping this grid to the canvas with positive die width and height in the prerendering module. 
The orientation does not influence this grid at all, the origin location will influence the direction in which they get drawn on the canvas (left -> right, top->down for example), but the virtual die grid is not influenced.
As they are represented as points, they do not have a width and a height, so in the case where the min and max are equal nothing is displayed.